### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by sundayezeilo**
This is a TDD (Test Driven Development) version of our Ruby's Enumerable Methods Project. RSpec version is implemented and documented in the readme file. We, therefore, request your review and comments so we can improve on our work. Thank you!

### CODE REVIEW

File: spec/enumerable_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        expect(example_array.my_select.class).to eql(example_array.select.class)
+      end
+
+      it 'returns a range enumerator when called on a range' do
+        expect(example_range.my_select.class).to eql(example_range.select.class)
+      end
+    end
+  end
+
+  describe '#my_all?' do
```
Comment (shshamim63) [inline spec/enumerable_methods_spec.rb:75]: Can you add a test where my_all? related to regex, of a certain class?

File: spec/enumerable_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        expect(example_array.my_all?).to eql(example_array.all?)
+      end
+
+      it 'returns true if all element in the range are truthy; otherwise, false' do
+        expect(example_range.my_all?).to eql(example_range.all?)
+      end
+    end
+  end
+
+  describe '#my_any?' do
```
Comment (shshamim63) [inline spec/enumerable_methods_spec.rb:127]: Can you add a test where my_any? related to regex, of a certain class?

File: spec/enumerable_methods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        expect(example_array.my_any?).to eql(example_array.any?)
+      end
+
+      it 'returns true if any element in the range are truthy; otherwise, false' do
+        expect(example_range.my_any?).to eql(example_range.any?)
+      end
+    end
+  end
+
+  describe '#my_none?' do
```
Comment (shshamim63) [inline spec/enumerable_methods_spec.rb:179]: Can you add a test where my_none? related to regex, of a certain class?

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :recycle: Required Changes :red_envelope: 
___
Hi @ezeilo-su and @Cathella :wave: 
You have gathered a lot of tests. But I think you have left some methods. Let's add them.
___
### :spades: Mandatory Changes :spades: 
___
1. Add some test for my_all?, my_any?, my_none?
2. Add test for my_each_with_index
3. Add some test for my_inject
___
##### Please add the following changes and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Review by serdg0 — APPROVED** [review]
**Congratulations** @Cathella @ezeilo-su  :fireworks: :100: :fireworks:
Feel free to merge.

Great work writing those tests!

Keep the good work in your next project!
**Happy Coding!** :computer: :nerd_face:
**_Sergio Diaz_**

**Pull request merged**